mono/metadata/cominterop.c

```diff
@@ -378,21 +378,18 @@ cominterop_get_method_interface (MonoMethod* method)
 		}
 	}
 
-	if (!ic) 
-		g_assert (ic);
-	g_assert (MONO_CLASS_IS_INTERFACE (ic));
-
 	return ic;
 }
 
 /**
  * cominterop_get_com_slot_for_method:
  * @method: a method
+ * @error: pointer to a bool indicating whether the lookup was successful or not; all failures will be considered fatal if it's set to null.
  *
  * Returns: the method's slot in the COM interface vtable
  */
 static int
-cominterop_get_com_slot_for_method (MonoMethod* method)
+cominterop_get_com_slot_for_method (MonoMethod* method, gboolean* error)
 {
 	guint32 slot = method->slot;
```

Comment:
This should use a MonoError.

Reply:
I've switched to MonoError, please let me know if there's something wrong with my implementation.